submissions/AnnaStoyano/frogger/app.js

```diff
@@ -0,0 +1,134 @@
+const canvas = {
+    width: 505,
+    height: 450
+}
+
+const Entity = function (x, y, speed, sprite) {
+    this.position = {
+        x,
+        y
+    };
+    this.speed = speed;
+    this.sprite = sprite;
+    this.level = 1;
+}
+
+Entity.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.position.x, this.position.y);
+}
+
+Entity.prototype.increaseLevel = function () {
+    this.level++;
+}
+
+Entity.prototype.checkWallCollision = function (wallX, wallY, newX, newY) {
+    if (this.position.x >= wallX) {
+        this.position.x = newX;
+    } else if (this.position.y >= wallY) {
+        this.position.y = newY;
+    }
+}
+
+/*Enemy*/
+
+const Enemy = function (x, y, speed, sprite, player) {
+    Entity.call(this, x, y, speed, sprite);
+    this.player = player;
+}
+
+Enemy.prototype = Object.create(Entity.prototype);
+Enemy.prototype.update = function (dt) {
+    this.position.x += this.speed * dt * (this.level * 0.8);
+    this.checkWallCollision(canvas.width, canvas.height, -90, 0);
+    this.checkLose();
+}
+
+Enemy.prototype.checkCollision = function () {
+    let positionX = Math.floor(this.position.x);
+    if (positionX - 50 < player.position.x && player.position.x < positionX + 50) {
```

Comment:
Here `player` is a global variable.

From the checklist:
> classes do not refer to any global variables, like global variable player, which is an instance of Player class
(referring to global constants and globals provided by the gaming platform like Resources is OK);

Double check your entire code base to meet the requirement.